client/core/core.go

```diff
@@ -1452,7 +1452,7 @@ func (c *Core) loadWallet(dbWallet *db.Wallet) (*xcWallet, error) {
 	walletCfg := &asset.WalletConfig{
 		Settings: dbWallet.Settings,
 		TipChange: func(err error) {
-			c.tipChange(assetID, err)
+			go c.tipChange(assetID, err)
```

Comment:
The asset backend should not need this callback to be anything in particular let alone synchronous.

Reply:
Should we let the asset backend decide? Call `go {dcr/btc}.tipChange` so it's more immediately obvious that the asset backend is not relying on the configured callback to finish processing before proceeding.

Reply:
Good points.  I had gone back and forth on this actually.  You make a good case, but my concern is that we then rely on the asset author to avoid deadlocking Core.  I'm not decided yet.

Reply:
Avoiding a deadlock in Core wouldn't be the main goal though, just a nice side effect. The main goal I see is making `monitorBlocks/checkForNewBlocks` faster, not waiting on the opaque execution of a callback, which could take however long _and_ which we really don't need to wait for.

Reply:
I agree with you.  There are also a million other ways the backend can make Disconnect hang indefinitely, so it's up to the consumer (Core) to deal.

Reply:
I've resolved this by wrapping the provided callback in the constructor rather than putting `go` in front over every place `tipChange` is called.  My reasoning is that there is no case where it would need to be synchronous and the programmer might forget. Although it does make it less obvious at the call site that it is asynchronous, but it is completely opaque so I don't think that says much to the reader anyway.